### PR TITLE
Improve eslint & add eslint-plugin-react-hooks

### PR DESCRIPTION
**Pull request by andypants888**
- checked eslint config
- tested on cli
- fixed syntax for 'max-len' rule
- added ignores for strings, urls
- (automatic) plugin version increased?
- removed unused import, AuthModal

### CODE REVIEW

**Comment by oriooctopus** [comment]
More descriptive title: something like "Improvements to eslint plugin" or even better "Improvements to eslint and add eslint-plugin-react-hooks"

**Comment by oriooctopus** [comment]
The description is great though, nice job on that

File: .eslintrc.js [hunk 1]
```diff
@@ -27,6 +28,9 @@ module.exports = {
   rules: {
     'react/react-in-jsx-scope': 'off',
     'unused-imports/no-unused-imports': 'error',
-    'max-len': 80,
+    'max-len': [
+      'warn',
+      { code: 80, tabWidth: 4, ignoreUrls: true, ignoreStrings: true },
```
Comment (oriooctopus) [inline .eslintrc.js]: shouldn't the tab width be 2?

**Review by oriooctopus — APPROVED** [review]

**Comment by oriooctopus** [comment]
Approved with some nits

**Review by andypants888 — COMMENTED** [review]
changed default eslint tabWidth 4 --> 2

**Pull request merged**